code/Weapons/Weapon.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Sandbox;
+
+namespace TerryForm.Weapons
+{
+	public partial class Weapon : BaseWeapon
+	{
+		public virtual string WeaponName => "";
+		public virtual string ModelPath => "weapons/rust_pumpshotgun/rust_pumpshotgun.vmdl";
+		public override float PrimaryRate => 2f;
+
+		public override void Spawn()
+		{
+			base.Spawn();
+
+			SetModel( ModelPath );
+		}
+
+		public override void ActiveStart( Entity ent )
+		{
+			base.ActiveStart( ent );
+
+			OnActiveEffects();
+		}
+
+		public override void ActiveEnd( Entity ent, bool dropped )
+		{
+			base.ActiveEnd( ent, dropped );
+
+			OnActiveEndEffects();
+		}
+
+		public override void Simulate( Client player )
+		{
+			base.Simulate( player );
+		}
+
+		public override bool CanPrimaryAttack()
+		{
+			var baseResult = base.CanPrimaryAttack();
+
+			//TODO: Check if it's my turn, if it isn't my turn don't allow me to shoot.
+			var myTurn = true;
+
+			if ( baseResult && myTurn )
+				return true;
+
+			return false;
+		}
+
+		public override void AttackPrimary()
+		{
+			OnFireEffects();
+
+			if ( !IsServer ) return;
```

Comment:
We should probably always put these on newlines, just for consistency's sake (like you did on lines 44-45):
```
if ( !IsServer )
    return;
```